### PR TITLE
Veena/expose mdm id in device info

### PR DESCRIPTION
**Pull request by Veena11**
## Proposed changes

Expose mdm id from sso config in device info sso request  for managed apps and 1P apps

## Type of change

- [ x] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ x] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: IdentityCore/src/broker_operation/response/MSIDDeviceInfo.m [hunk 1]
```diff
@@ -77,7 +78,7 @@ - (NSDictionary *)jsonDictionary
     json[MSID_BROKER_WPJ_STATUS_KEY] = [self wpjStatusStringFromEnum:self.wpjStatus];
     json[MSID_BROKER_BROKER_VERSION_KEY] = self.brokerVersion;
     json[MSID_ADDITIONAL_EXTENSION_DATA_KEY] = [self.additionalExtensionData msidJSONSerializeWithContext:nil];
-    
+    json[MSID_BROKER_MDM_ID_KEY] = self.mdmId;
```
Comment (kaisong1990) [inline IdentityCore/src/broker_operation/response/MSIDDeviceInfo.m:81]: Do we still assign if mdmid is nil?

**Review by kaisong1990 — APPROVED** [review]
LGTM, just a monor question

File: IdentityCore/src/broker_operation/response/MSIDDeviceInfo.m (hunk 1 above)
Comment (Veena11) [reply IdentityCore/src/broker_operation/response/MSIDDeviceInfo.m:81]: Do we always look for not null before assigning to json dict ? I wasnt sure

Comment (Veena11) [reply IdentityCore/src/broker_operation/response/MSIDDeviceInfo.m:81]: Added the nil check before updating extraDeviceInformation in MSAL

**Pull request merged**